src/protocol.c

```diff
@@ -102,9 +102,34 @@ static bool spawn_process(struct pss_tty *pss, uint16_t columns, uint16_t rows)
   for (i = 0; i < server->argc; i++) {
     argv[n++] = server->argv[i];
   }
-  for (i = 0; i < pss->argc; i++) {
-    argv[n++] = pss->args[i];
+  if (server->url_arg) {
+    for (i = 0; i < pss->argc; i++) {
+      argv[n++] = pss->args[i];
+    }
   }
+  else if (server->arg_file != NULL) {
+    int fd = -1;
+    // mkstemp requires the file path to have suffix XXXXXX (len 7)
```

Comment:
XXXXXX is len 6. no?

Reply:
```
int file_path_len = strlen(server->arg_file) + 6 /*XXXXXX*/ + 1 /*null character*/;
```

Reply:
(subjective opinion)
maybe strcpy,strcat might be "simpler" to follow 
```
strcpy(filePath, server->arg_file);
strcat(filePath, "XXXXXX");
```

Reply:
Yeah `XXXXXX` has len 6, but with the null terminator it's 7. I'll change this line to be more clear.

Reply:
I was using `strcat` before but it might not be safe: https://github.com/tsl0922/ttyd/pull/747#discussion_r684435038